pkg/resources/istioingress/istioingress.go

```diff
@@ -70,39 +83,82 @@ func New(client client.Client, cluster *v1beta1.KafkaCluster) *Reconciler {
 func (r *Reconciler) Reconcile(log logr.Logger) error {
 	log = log.WithValues("component", componentName)
 	log.V(1).Info("Reconciling")
-	if r.KafkaCluster.Spec.GetIngressController() == istioingress.IngressControllerName {
-		for _, eListener := range r.KafkaCluster.Spec.ListenersConfig.ExternalListeners {
-			if eListener.GetAccessMethod() == corev1.ServiceTypeLoadBalancer {
-				if r.KafkaCluster.Spec.IstioControlPlane == nil {
-					log.Error(errors.NewPlain("reference to Istio Control Plane is missing"), "skip external listener reconciliation", "external listener", eListener.Name)
+
+	for _, eListener := range r.KafkaCluster.Spec.ListenersConfig.ExternalListeners {
+		if r.KafkaCluster.Spec.GetIngressController() == istioingress.IngressControllerName && eListener.GetAccessMethod() == corev1.ServiceTypeLoadBalancer {
+			if r.KafkaCluster.Spec.IstioControlPlane == nil {
+				log.Error(errors.NewPlain("reference to Istio Control Plane is missing"), "skip external listener reconciliation", "external listener", eListener.Name)
+				continue
+			}
+
+			istioRevision := istioOperatorApi.NamespacedRevision(
+				strings.ReplaceAll(r.KafkaCluster.Spec.IstioControlPlane.Name, ".", "-"),
+				r.KafkaCluster.Spec.IstioControlPlane.Namespace)
+			ingressConfigs, defaultControllerName, err := util.GetIngressConfigs(r.KafkaCluster.Spec, eListener)
+			if err != nil {
+				return err
+			}
+			for name, ingressConfig := range ingressConfigs {
+				if !util.IsIngressConfigInUse(name, defaultControllerName, r.KafkaCluster, log) {
 					continue
 				}
+				for _, res := range []resources.ResourceWithLogAndExternalListenerSpecificInfosAndIstioRevision{
+					r.meshgateway,
+					r.gateway,
+					r.virtualService,
+				} {
+					o := res(log, eListener, ingressConfig, name, defaultControllerName, istioRevision)
+					err := k8sutil.Reconcile(log, r.Client, o, r.KafkaCluster)
+					if err != nil {
+						return err
+					}
+				}
+			}
+		} else {
+			// Cleaning up unused istio resources when ingress controller is not istioingress or eListener access method is not LoadBalancer
+			deletionCounter := 0
+			ctx := context.Background()
+			istioResourcesGVK := []schema.GroupVersionKind{
+				{
+					Version: istioOperatorApi.GroupVersion.Version,
+					Group:   istioOperatorApi.GroupVersion.Group,
+					Kind:    reflect.TypeOf(istioOperatorApi.IstioMeshGateway{}).Name(),
+				},
+				{
+					Version: istioclientv1beta1.SchemeGroupVersion.Version,
+					Group:   istioclientv1beta1.SchemeGroupVersion.Group,
+					Kind:    reflect.TypeOf(istioclientv1beta1.Gateway{}).Name(),
+				},
+				{
+					Version: istioclientv1beta1.SchemeGroupVersion.Version,
+					Group:   istioclientv1beta1.SchemeGroupVersion.Group,
+					Kind:    reflect.TypeOf(istioclientv1beta1.VirtualService{}).Name(),
+				},
+			}
+			var istioResources unstructured.UnstructuredList
```

Comment:
Again, can't see this being used anywhere outside the cycle, can't we just define it inside?